### ISSUE TITLE
Should be possible to make and run without install

### ISSUE THREAD

**Issue by NiKiZe** [firehol/netdata#149]
To test software and sometimes for security reasons, It is good practice to just compile sources and then run without clobbering up your system directories.

Having a script that does "magic things" is something most don't want to run, especially since it is not really documented what it actually does.

So after cloning, running autogen && ./configure && make
we run src/main ... netdata: Cannot open file '/usr/local/etc/netdata/netdata.conf'
well ofcourse not, I didn't install, please simply run everything from local directory as is.
The same goes for logs etc. Assume it don't have root access (it shoul never have root access to begin with) and sen logs thru syslog instead etc.

**Comment by ktsaou** [comment]
Why don't you install it like this:

```sh
./netdata-installer.sh --install /tmp
```

Then delete the directory `netdata` from `/tmp`.

If you do it like that, you will also have to delete the `netdata` user and group to completely undo what the installer did.

**Comment by NiKiZe** [comment]
"If you do it like that, you will also have to delete the netdata user and group to completely undo what the installer did."

How can a group be created when the installer won't be run as root?

**Comment by ktsaou** [comment]
No it cannot.
However, parts of netdata require root access to work (like `apps.plugin`). So, we advise our users to install it as root.

What I normally do, is that I install non-system apps in `/opt` and I keep the there. This allows me to isolate them from the rest of the system.

If you just want to install to view it, the demo site is pretty much what you will get. The demo site is a netdata installation and usually is worst from what will get when you install it (more users on it, bigger network latency). It is however the real thing you are going to install.

**Comment by NiKiZe** [comment]
I do wan't to run it, (I want to see my own data) but I'm on the paranoid side and don't trust anything, so I want to run it as a separate user and not as root, that goes for all parts of it (including install, which I wont run at all)

I think there is others with similar security policies in place. so that "requirement" for netdata is a big drawback. :(

**Comment by ktsaou** [comment]
Unfortunately I will disagree.

netdata is a monitoring app. It needs access to the thing it will be monitoring. If you cannot give it access, it cannot monitor it.

This is not a requirement of netdata. The Linux kernel imposes these restrictions and hopefully no application can do otherwise.

Anyway, to build and install netdata in /tmp, do this:

```
./autogen.sh
./configure --prefix="/tmp/netdata/usr" --sysconfdir="/tmp/netdata/etc" --localstatedir="/tmp/netdata/var" --with-zlib --with-math --with-user=$USERNAME
make
make install
```

The last command will run `sudo` to setuid to root `apps.plugin`. I suggest to do it in order to complete the installation to /tmp. Then, delete `apps.plugin`:

```
rm /tmp/netdata/usr/libexec/netdata/plugins.d/apps.plugin
```

Now run netdata with:

```
/tmp/netdata/usr/sbin/netdata
```

It will print a few errors. Ignore them. It runs.

Using this setup applications are not monitored. If you really want to allow `apps.plugin` to run as root, do not delete it.

Once done, delete `/tmp/netdata`.

**Comment by NiKiZe** [comment]
As anyone should know, especially linux folks, runing as root is bad, no excpetions.
Is this the official stance for the project? If so, I'm sorry to hear that netdata does not take security seriously.

Feel free to close as "won't fix" if there won't be any "install and run as limited user" available.

@ktsaou thanks for you time, rally wish I could test it since it looks so great, but security must go first.

**Comment by ktsaou** [comment]
I am sorry, I don't understand.

netdata is not running as root. Even if started as root, it switches to another user.

Only apps.plugin is running as root because it cannot do otherwise. It needs to read the entire `/proc` filesystem. If you know a way to avoid this, please let us know. I haven't found one. If I had, that way would have been used.

**Comment by ktsaou** [comment]
I have updated my answer above, so read it on github.

**Comment by ktsaou** [comment]
With the latest commit, the installer is able to build and install netdata without root access.
The switch `apps.plugin` from setuid to capabilities is not yet implemented ( #157 ). You can follow up this progress there.

If the issue is resolved, please close this.

**Comment by ktsaou** [comment]
Cleaning up. If something remains, please open it again.
